runtime/context.go

```diff
@@ -207,6 +209,19 @@ func accumulateLogFields(ctx context.Context, newFields []zap.Field) []zap.Field
 	return append(previousFields, newFields...)
 }
 
+func accumulateLogMsgAndFieldsInContext(ctx context.Context, msg string, newFields []zap.Field) context.Context {
+	ctxLogCounter := 1
+	v := ctx.Value(ctxLogCounterName)
+	if v != nil {
+		ctxLogCounter = v.(int)
+		ctxLogCounter++
```

Comment:
Please **confirm** if this is in the scope of a single request and atomic update is not required.

Reply:
separate requests maintain separate contexts, so this should work fine.